spek-runtimes/common/src/main/kotlin/org/spekframework/spek2/runtime/scope/Path.kt

```diff
@@ -18,3 +18,9 @@ val Path.isRoot: Boolean
 // 1: my.package/MyClass -> my.package/MyClass/description
 // 2: my.package/MyClass/description -> my.package/MyClass
 fun Path.isRelated(path: Path) = this.isParentOf(path) || path.isParentOf(this)
+
+
+interface PathBuilder {
+    fun append(name: String): PathBuilder
```

Comment:
Maybe we should have `fun package()`, `fun class()`, etc instead of super generic `append()`? I'm just afraid such generic `String` based api will be easy to misuse :)

Reply:
Wouldn't that just complicate things? I would have to add intermediate interfaces:

```kotlin
interface ClassPathBuilder {
    fun scope(scope: String): ClassPathBuilder
    fun builder(): Path
}

interface PackagePathBuilder {
    fun class(cls: String): ClassPathBuilder
}

interface PathBuilder {
    fun package(pkg: String):  PackagePathBuilder
}
```
(ignore the naming for now)

Reply:
Well, okay, but some kind of validation of the resulting string will be great then

Anyway, this is internal API so we can change it if it'll cause some problems :)

Reply:
What do you mean by validation?

Reply:
That path points to some valid thing like class, or method because atm it's just a string builder :)

Reply:
It won't contain just a package and a class, it will also contain the scope descriptions. `com.example/SomeSpek/describe something/it should do this`.